machine/v1/types_nutanixprovider.go

```diff
@@ -66,6 +84,29 @@ type NutanixMachineProviderConfig struct {
 	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret"`
 }
 
+// NutanixCategory identifies a pair of prism category key and value
+type NutanixCategory struct {
+	// +kubebuilder:validation:Required
+	Key string `json:"key"`
+
+	// +kubebuilder:validation:Required
+	Value string `json:"value"`
```

Comment:
Can you please add godoc to explain these two fields

Also, are there any validations we can put on, for example, a maximum length or a set of valid characters?

Reply:
Addressed. Only found that both the category key and value requires <= 64 characters, but not found any restriction on the characters pattern. Also updated the godoc of the requirement that all the categories specified in the Machine spec need to already exist (defined) in the prism central.